src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxePsiClass.java

```diff
@@ -72,15 +72,29 @@ public HaxeNamedComponent getTypeComponent() {
   @NotNull
   @Override
   public String getQualifiedName() {
-    final String name = getName();
+    String name = getName();
     if (getParent() == null) {
       return name == null ? "" : name;
     }
+
+    if(name == null && this instanceof HaxeAnonymousType) {
+      // restore name from parent
+      HaxeTypedefDeclaration typedefDecl = (HaxeTypedefDeclaration)(getParent().getParent());
+      if(typedefDecl != null) {
+        name = typedefDecl.getName();
+      }
+    }
+
     final String fileName = FileUtil.getNameWithoutExtension(getContainingFile().getName());
     String packageName = HaxeResolveUtil.getPackageName(getContainingFile());
-    if (isAncillaryClass(name, fileName)) {
+
+    // Check type is `Ancillary` if:
+    // 1. NOT root package! (packageName is NOT empty)
+    // 2. `name` is not null
+    if (name != null && !packageName.isEmpty() && isAncillaryClass(name, fileName)) {
```

Comment:
Apologize, only StdTypes module is global, so we need to check only `packageName.isEmpty() && fileName == "StdTypes"`